### PR TITLE
Deploy lunar-hotel-api on render

### PR DESCRIPTION
**Pull request by medaminedev66**
In this pull request, I implemented the following changes:
- Updated the Node version to 20.10.0, as specified in the .node-version file.
- Updated the Ruby version to 3.0.1.
- Removed unnecessary webpack files, considering the application's focus on API and its lack of reliance on the webpack compiler.
- Introduced new files to facilitate Render deployment:
  - render-build.sh to execute Rails commands for production.
  - render.yaml file for configuring and establishing a connection with the database.
- Updated the MASTER_KEY and credential.yml for enhanced security.

### CODE REVIEW

File: .gitignore
Comment (mmatongo) [inline .gitignore:1]: Logfiles should ideally be kept within the application's instance for easier debugging.

File: .browserslistrc
Comment (mmatongo) [inline .browserslistrc:1]: Because this is an api this is not necessary.

File: .node-version
Comment (mmatongo) [inline .node-version:1]: This is a frontend artifact

File: Gemfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 source 'https://rubygems.org'
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
-ruby '2.7.5'
+ruby File.read('.ruby-version').strip
```
Comment (mmatongo) [inline Gemfile:4]: Abstractions like this should be avoided as they lead to unintended sideeffects

File: bin/render-build.sh
Comment (mmatongo) [inline bin/render-build.sh:1]: ~A better approach would be doing this with a Procfile~
Render seems to prefer this

File: config/database.yml
Comment (mmatongo) [inline config/database.yml:1]: Avoid hardcoding credentials into config files and use the credentials.yml for this

File: config/environments/production.rb
Comment (mmatongo) [inline config/environments/production.rb:1]: Should this line even be present?

File: package.json
Comment (mmatongo) [inline package.json:1]: The package and package-lock files should ideally be removed

File: .node-version
Comment (medaminedev66) [reply .node-version:1]: The deployment on Render faced an issue that got resolved after updating the Node version. Although I'm not certain about the exact cause, it's possible that the webpack and package.json files in the application were involved.

File: Gemfile (hunk 1 above)
Comment (medaminedev66) [reply Gemfile:4]: Thank you, @mmatongo. I will correct that.
I was following this guide: https://docs.render.com/ruby-version

File: package.json
Comment (medaminedev66) [reply package.json:1]: This app utilized Swagger, and currently, I'm unsure whether the package.json file is necessary or not.

**Comment by medaminedev66** [comment]
@mmatongo The live demo link is available now. Could you review it?
Here is the link: https://lunar-hotel.onrender.com/

File: package.json
Comment (mmatongo) [reply package.json:1]: rswag uses the rswag-ui gem if I am correct so I think we can do without the files.

File: Gemfile (hunk 1 above)
Comment (mmatongo) [reply Gemfile:4]: Odd that they would recommend that

**Comment by mmatongo** [comment]
> @mmatongo The live demo link is available now. Could you review it? Here is the link: https://lunar-hotel.onrender.com/

LGTM

**Pull request merged**